pkg/scheduler/testing/wrappers.go

```diff
@@ -285,6 +297,17 @@ func (p *PodWrapper) HostPort(port int32) *PodWrapper {
 	return p
 }
 
+// ContainerPorts creates a container with ports valued `ports` if inner has no existing containers,
+// otherwise it will set the first container ports to be `ports. for the inner pod.
+func (p *PodWrapper) ContainerPorts(ports []v1.ContainerPort) *PodWrapper {
+	if len(p.Spec.Containers) == 0 {
+		p.Spec.Containers = append(p.Spec.Containers, v1.Container{})
+	}
+
+	p.Spec.Containers[0].Ports = ports
```

Comment:
Why not the all Containers?

Reply:
Changed to allow a container to be created and then assign the ports to the new container, thanks for catching this.